application/org.openjdk.jmc.flightrecorder.ui/src/main/java/org/openjdk/jmc/flightrecorder/ui/pages/ConstantPoolsPage.java

```diff
@@ -0,0 +1,195 @@
+		private ItemHistogramWithInput byTypeHistogram;
+		private ItemHistogramWithInput constantHistogram;
+
+		ConstantPoolsPageUi(Composite parent, FormToolkit toolkit, IPageContainer pageContainer, IState state) {
+			constPoolItems = getDataSource().getConstantPools();
+			selectionConstPoolItems = constPoolItems;
+			selectionConstantItems = getDataSource().getConstants();
+			Form form = DataPageToolkit.createForm(parent, toolkit, getName(), getIcon());
+			sash = new SashForm(form.getBody(), SWT.VERTICAL);
+			toolkit.adapt(sash);
+			byTypeTable = BY_TYPE_CONSTANT_POOLS_HISTOGRAM.buildWithoutBorder(sash, JdkAttributes.CONSTANT_POOLS_NAME,
+					new TableSettings(state.getChild(TABLE)));
+			MCContextMenuManager mm = MCContextMenuManager.create(byTypeTable.getManager().getViewer().getControl());
+			ColumnMenusFactory.addDefaultMenus(byTypeTable.getManager(), mm);
+			byTypeFilter = FilterComponent.createFilterComponent(byTypeTable, null, constPoolItems,
+					pageContainer.getSelectionStore()::getSelections, this::onTypeFilterChange);
+			//mm.add(byTypeFilter.getShowFilterAction());
+			mm.add(byTypeFilter.getShowSearchAction());
+
+			constantValueTable = CONSTANT_HISTOGRAM.buildWithoutBorder(sash, JdkAttributes.CONSTANT_VALUE,
+					new TableSettings(state.getChild(CONSTANT_TABLE)));
+			mm = MCContextMenuManager.create(constantValueTable.getManager().getViewer().getControl());
+			ColumnMenusFactory.addDefaultMenus(constantValueTable.getManager(), mm);
+			constantValueFilter = FilterComponent.createFilterComponent(constantValueTable, null,
+					getDataSource().getConstants(), pageContainer.getSelectionStore()::getSelections,
+					this::onValueFilterChange);
+			//mm.add(constantValueFilter.getShowFilterAction());
+			mm.add(constantValueFilter.getShowSearchAction());
+
+			// chains the 2 tables
+			byTypeHistogram = new ItemHistogramWithInput(byTypeTable);
+			constantHistogram = new ItemHistogramWithInput(constantValueTable);
+			byTypeHistogram.addListener(this::typeHistogramListener);
+
+			byTypeFilter.loadState(state.getChild(TYPE_FILTER));
+			constantValueFilter.loadState(state.getChild(VALUE_FILTER));
+		}
+
+		private void typeHistogramListener(IItemCollection constantPoolType) {
+			String poolName = ItemToolkit.getFirstFound(constantPoolType, JdkAttributes.CONSTANT_POOLS_NAME);
+			IItemCollection filteredItems = getDataSource().getConstants()
+					.apply(ItemFilters.equals(JdkAttributes.CONSTANT_TYPE, poolName));
+			constantHistogram.setInput(filteredItems);
+		}
+
+		private void onTypeFilterChange(IItemFilter filter) {
+			byTypeFilter.filterChangeHelper(filter, byTypeTable, selectionConstPoolItems);
+		}
+
+		private void onValueFilterChange(IItemFilter filter) {
+			constantValueFilter.filterChangeHelper(filter, constantValueTable, selectionConstantItems);
+		}
+
+		@Override
+		public void saveTo(IWritableState writableState) {
+			PersistableSashForm.saveState(sash, writableState.createChild(SASH_ELEMENT));
```

Comment:
This saves the state of the SashForm, but that state isn't loaded in the IPageUI constructor, so the value used will only be the default for a SashForm.

E.g. `PersistableSashForm.loadState(sash, state.getChild(SASH));`

Reply:
done